commands/decimal_spiral.go

```diff
@@ -0,0 +1,258 @@
+package commands
+
+import (
+	"errors"
+	"strconv"
+
+    "github.com/bwmarrin/discordgo"
+
+    "github.com/unswpcsoc/PCSocBot/utils"
+)
+
+const(
+    DecimalSpiralLowerLimit = 5
+    DecimalSpiralUpperLimit = 43
+)
+
+var (
+	ErrDecimalSpiralSyntax = errors.New("size is not an integer")
+	ErrDecimalSpiralRange = errors.New("size is not an odd integer between " + strconv.Itoa(DecimalSpiralLowerLimit) + " and " + strconv.Itoa(DecimalSpiralUpperLimit))
+	ErrDecimalSpiralArgs = errors.New("not enough args")
+)
+
+type DecimalSpiral struct {
+	names []string
+	desc string
+}
+
+func NewDecimalSpiral() *DecimalSpiral {
+	return &DecimalSpiral{
+		names: []string{"decimalspiral", "ds"},
+		desc: "Usage: ds <size>. Generate a decimal spiral. Size must be an odd integer between " + strconv.Itoa(DecimalSpiralLowerLimit) + " and " + strconv.Itoa(DecimalSpiralUpperLimit),
+	}
+}
+
+func (d *DecimalSpiral) Names() []string {
+	return d.names
+}
+
+func (d *DecimalSpiral) Desc() string {
+	return d.desc
+}
+
+func (d *DecimalSpiral) Roles() []string {
+	return nil
+}
+
+func (d *DecimalSpiral) Chans() []string {
+	return nil
+}
+
+func (d *DecimalSpiral) MsgHandle(ses *discordgo.Session, msg *discordgo.Message, args []string) (*CommandSend, error) {
+	if len(args) == 0 {
+		return nil, ErrDecimalSpiralArgs
+	}
+
+    size, err := strconv.Atoi(args[0]);
+    if err != nil {
+        if err.(*strconv.NumError).Err == strconv.ErrSyntax {
+            return nil, ErrDecimalSpiralSyntax
+        } else if err.(*strconv.NumError).Err == strconv.ErrRange {
+            return nil, ErrDecimalSpiralRange
+        } else {
+            return nil, err
+        }
+    }
+
+	if isEven(size) || size < DecimalSpiralLowerLimit || size > DecimalSpiralUpperLimit {
```

Comment:
Avoid making functions that could be put on one line. We teach you to make simple functions in 1511 to help grasp the concept of abstractions. However, low-level abstractions like this only add calling overhead (setting up and taking down a stack frame for the function).
```suggestion
	if size % 2 == 0 || size < DecimalSpiralLowerLimit || size > DecimalSpiralUpperLimit {
```

Reply:
Oh, thank you for this tip!